src/WebJobs.Script.WebHost/WebScriptHostExceptionHandler.cs

```diff
@@ -48,8 +54,15 @@ public async Task OnTimeoutExceptionAsync(ExceptionDispatchInfo exceptionInfo, T
             // Task ignoreTask = _hostManager.StopAsync();
             // Give the manager and all running tasks some time to shut down gracefully.
             //await Task.Delay(timeoutGracePeriod);
-
-            _applicationLifetime.StopApplication();
+            string workerRuntime = _environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerRuntimeSettingName);
+            if (string.IsNullOrEmpty(workerRuntime) || string.Equals(workerRuntime, RpcWorkerConstants.DotNetLanguageWorkerName, StringComparison.InvariantCultureIgnoreCase))
+            {
+                _applicationLifetime.StopApplication();
+            }
+            else
+            {
+                await _functionInvocationDispatcher.RestartAsync();
```

Comment:
Add warning level log indicating Jobhost is restarting

Reply:
Will do. But also, this is not a jobhost restart.. Just want to make sure we're on the same page here :)